### ISSUE TITLE
Drupal root incorrect on subdomain, even with site alias.

### ISSUE THREAD

**Issue by ItsFrankieD** [drush-ops/drush#5387]
**Describe the bug**
I've installed a Drupal 10 site alongside my primary Drupal 9 install. To note, I haven't had this issue prior to Drupal 10 with other subdomains.

Running Drush through my primary domain located in my home directory works as expected. `drush status` returns everything you'd expect. Running in my subdomain's directory, though, produces the wrong Drupal root directory and fails to run, stating that it cannot find the root.

Running Drush in my primary domain's directory returns `Drupal root : /home/username/public_html`, which is the expected result. Running it under my subdomain returns `Drupal root : /home/username/dev/public_html/web`, whereas I'm expecting `/home/username/dev/public_html`. For the life of me, I can't find where the "/web" is coming from, and it's nowhere in my configs. This has rendered development on my subdomain unmanageable.

I have verified the following alias file has been found, but it produces the same result.
```
# File: dev.site.yml
dev:
  root: /home/username/dev/public_html
  uri: https://dev.example.com
```

This goes without saying, but I've subbed in placeholder names for my directories.

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 11.5.0
| Drupal version? | 10.0.3
| PHP version     | 8.1.6
| OS?             | Linux

I'm willing to provide any information necessary to help diagnose, as this has brought my module development process to a near halt. Thanks much.

**Comment by greg-1-anderson** [comment]
Looks like a simple typo. The `root` element in the site alias you list above has the value that you says is wrong. Try changing it to `/home/username/dev/www`, the value you say is correct, and see if that clears up the trouble.

**Comment by ItsFrankieD** [comment]
My apologies, I copied the wrong bit over to the issue. The issue is that Drush appends "/web" to the end of the path, Updated.

**Comment by greg-1-anderson** [comment]
Is there a `web` directory that exists? Does it contain an index.php? It could be a Drupal Finder issue.

**Comment by ItsFrankieD** [comment]
No, there's no `web` directory. Besides a custom module I was working on, it's a stock D10 install using `drupal/recommended-project`. The only things outside of webroot are my composer files and dependencies, the webroot itself, and an unrelated symlink.

**Comment by ItsFrankieD** [comment]
Any suggestion on where to go from here? Would the next step be to create an issue for `webflo/drupal-finder`?

**Comment by greg-1-anderson** [comment]
I don't think anyone but you is going to be able to debug your configuration issue. I think what I'd do is run `composer create-project drupal/recommended-project` and observe Drush working in that environment. Then, make that test site look more like your site, one step at a time, until you discover the difference that prevents Drush from finding your Drupal root.

You could also try running `drush` with `--debug` and see if any of the log messages give any clues, or step through the Drush bootstrap and observe where it is selecting the wrong root.

**Comment by weitzman** [comment]
The drupal-finder uses an entry in composer.json `extra.installer-paths.type:drupal-core` to find the root. Take a look there first.

**Comment by ItsFrankieD** [comment]
Following both of yours advice, I found the culprit. Any installer path for core besides `"www/core"` yielded the correct path, so I decided to give the web root one final glance. Lo and behold, there was a copy of the initial `drupal/recommended-project` JSON whose web root was set to `"web/core"`.

I hadn't even considered the possibility of Drupal Finder chaining JSONs. One oversight on my end turned what was intended to be a weekend project into a week-long search for the source of this issue.

Thanks for the help!
